### PR TITLE
fix: test_symlink

### PR DESCRIPTION
**Pull request by create2000**
Worked on disabling the windows symlink.

### CODE REVIEW

File: lib_eio_windows/test/test_fs.ml [hunk 1]
```diff
@@ -119,80 +114,61 @@ let test_append env () =
...
-  try_mkdir (cwd / "subdir");
-  try_mkdir (cwd / "subdir\\nested");
-  let test_file = cwd / "subdir\\nested\\test-file" in
-  Path.save ~create:(`Exclusive 0o600) test_file "data";
-  Alcotest.(check string) "mkdir" "data" (Path.load test_file);
-  Unix.unlink "subdir\\nested\\test-file";
-  Unix.rmdir "subdir\\nested";
-  Unix.rmdir "subdir"
-
-let test_mkdirs env () =
```
Comment (patricoferris) [inline lib_eio_windows/test/test_fs.ml:133]: As mentioned in the previous PR, could we add this test to the bottom of the file alongside the other tests instead of removing it. Thanks!

File: lib_eio_windows/test/test_fs.ml [hunk 2]
```diff
@@ -119,80 +114,61 @@ let test_append env () =
   Path.save ~create:`Never ~append:true test_file "2nd-write";
   Alcotest.(check string) "append" "1st-write-original2nd-write" (Path.load test_file)
 
-let test_mkdir env () =
```
Comment (patricoferris) [inline lib_eio_windows/test/test_fs.ml:122]: Why did you move `test_mkdir` from here to further down the file? It is an extra diff that doesn't seem to be necessary, but perhaps I am mistaken?

File: lib_eio_windows/test/test_fs.ml [hunk 3]
```diff
@@ -119,80 +114,61 @@ let test_append env () =
...
-  Unix.symlink ~to_dir:true "foo" "sandbox\\dangle";
-  try_mkdir (cwd / "tmp");
-  Eio.Path.with_open_dir (cwd / "sandbox") @@ fun sandbox ->
-  try_mkdir (sandbox / "subdir");
-  try_mkdir (sandbox / "to-subdir\\nested");
-  let () =
-    try
+
+  if not (Unix.has_symlink ()) then
+    Printf.printf "Skipping test_symlink on systems that don't support symlinks.\n"
```
Comment (patricoferris) [inline lib_eio_windows/test/test_fs.ml:135]: This looks good to me, thank you!

**Review by patricoferris — CHANGES_REQUESTED** [review]
The core fix is good, but we need to not also make other changes whilst doing it :))

File: lib_eio_windows/test/test_fs.ml (hunk 2 above)
Comment (create2000) [reply lib_eio_windows/test/test_fs.ml:122]: No, i think it must have been an oversight from me.

File: lib_eio_windows/test/test_fs.ml (hunk 3 above)
Comment (create2000) [reply lib_eio_windows/test/test_fs.ml:135]: Alright. Thanks.

**Comment by create2000** [comment]
@patricoferris -- I have done the fixes. Is this okay?

**Comment by patricoferris** [comment]
Definitely getting there @create2000.

I would recommend resetting this history and only (a) making the fix to `test_symlink` and [adding the test_mkdirs into the tests of this file](https://github.com/ocaml-multicore/eio/blob/216ccff845d68ada8c7d3bd31581cbf37e8ee791/lib_eio_windows/test/test_fs.ml#L268-L279).

**Comment by create2000** [comment]
Thank you @patricoferris -- I have added the ```test_mkdirs``` to the tests. About making the fix to ```test_symlink``` I understand it to mean resetting my git history to reflect `fix: test_symlink`. If this is it, I have done that.

**Comment by patricoferris** [comment]
Thanks @create2000 :))

> mean resetting my git history to reflect fix: test_symlink

My point here is that right now, the [diff](https://github.com/ocaml-multicore/eio/pull/768/files) is a little confusing. We have four distinct changes happening:
 1. The fix for the symlink test
 2. Adding `test_mkdirs` to the test suite
 3. White-spacing/formating changes
 4. The actual `test_mkdirs` function being moved from the top of the file to the bottom
 
 Ideally, this would only be changes (1) and (2). I'm okay with (3) slipping into this PR, especially since the codebase has no automatic formatting. However, (4) seems to be a change that is a resulting of first deleting that function and then rewriting it somewhere else. It would be good if the history wasn't like that.
 
 So concretely, I think we want to redo this PR with just (1), (2) and (3). Does that make sense?

**Comment by create2000** [comment]
Thank you @patricoferris . I understand now what you mean. I will close this PR and [here](https://github.com/ocaml-multicore/eio/pull/771) is the new PR created to keep everything clean.  I hope the new PR meets the corrections. 

Also, In recording my contribution on the Outreachy contribution page, I am asked to provide a URL of my contributions if it is merged or accepted. As this is still ongoing, can I still submit a URL in the hope that it will get merged before the contribution deadline? This is to keep with the advice of Outreachy to record our contributions early.
